sip-core/src/main/java/de/ikor/sip/foundation/core/trace/CustomTracer.java

```diff
@@ -18,35 +18,40 @@ public class CustomTracer extends DefaultTracer {
 
   public static final String TRACING_ID = "tracingId";
 
-  private final Set<SIPTraceOperation> sipTraceOperations;
-
-  private final TraceHistory traceHistory;
+  private final SIPTraceConfig sipTraceConfig;
 
   /**
    * Creates new instance of CustomTracer Enables tracing in CamelContext
    *
-   * @param traceHistory {@link TraceHistory}
    * @param exchangeFormatter {@link SIPExchangeFormatter}
    * @param camelContext {@link CamelContext}
-   * @param sipTraceOperations set of {@link SIPTraceOperation}
+   * @param sipTraceConfig set of {@link SIPTraceConfig}
    */
   public CustomTracer(
-      TraceHistory traceHistory,
       SIPExchangeFormatter exchangeFormatter,
       CamelContext camelContext,
-      Set<SIPTraceOperation> sipTraceOperations) {
+      SIPTraceConfig sipTraceConfig) {
     setExchangeFormatter(exchangeFormatter);
     camelContext.setTracing(true);
-    this.traceHistory = traceHistory;
-    this.sipTraceOperations = sipTraceOperations;
+    this.sipTraceConfig = sipTraceConfig;
+  }
+
+  @Override
+  public void traceBeforeRoute(NamedRoute route, Exchange exchange) {
+    setTracingID(exchange);
+    super.traceBeforeRoute(route, exchange);
   }
 
   @Override
   public void traceBeforeNode(NamedNode node, Exchange exchange) {
-    exchange.getIn().setHeader(TRACING_ID, tracingList(exchange));
+    setTracingID(exchange);
```

Comment:
the name should explain what happens
it does not set the ID, it adds it into some tracing id list